src/state/opamSysInteract.ml

```diff
@@ -253,31 +251,27 @@ let packages_status packages =
           | Some (pkg, _repo) -> pkg
           | None -> s)
       |> (fun l -> run_query_command "apk" ("policy"::l))
-      (* fst_version is here to do not add package while their informations
-         parsing it not yet finished, or at least a minimum complete *)
-      |> List.fold_left (fun (current, fst_version, installed, repo, instavail) l ->
-          try
+      |> List.fold_left (fun (pkg, installed, instavail) l ->
+          try (* package name *)
             let new_pkg = Re.(Group.get (exec pkg_name l) 1) in
-            let pkg = match repo with Some r -> current^"@"^r | None -> current in
-            let instavail = add_pkg fst_version pkg installed instavail in
-            new_pkg, true, false, None, instavail
+            new_pkg, false, instavail
           with Not_found ->
             if l.[2] != ' ' then (* only version in after two spaces *)
-              let pkg = match repo with Some r -> current^"@"^r | None -> current in
-              let instavail = add_pkg fst_version pkg installed instavail in
-              current, false, false, None, instavail
-            else
-            if Re.execp is_installed l then
-              current, fst_version, true, repo, instavail
-            else
-            try
-              let grs = Re.exec repo_name l in
-              let repo = Some (Re.Group.get grs 1) in
-              current, fst_version, installed, repo, instavail
-            with Not_found ->
-              current, fst_version, installed, repo, instavail)
-        ("", true, false, None, OpamSysPkg.Set.(empty, empty))
-      |> (fun (_,_,_,_, instavail) -> instavail)
+              let installed =
+                OpamStd.String.ends_with ~suffix:"(installed):" l
```

Comment:
According to https://git.alpinelinux.org/apk-tools/tree/src/app_policy.c and local testing this is never the case. Their own example is wrong and probably just an approximation